tests/Extensions/Hosting.Services.UnitTests/HostBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Fabric;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.Omex.Extensions.Hosting.Services;
+using Microsoft.Omex.Extensions.Logging;
+using Microsoft.Omex.Extensions.TimedScopes;
+using Microsoft.ServiceFabric.Services.Communication.Runtime;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using ServiceFabric.Mocks;
+
+namespace Hosting.Services.UnitTests
+{
+	[TestClass]
+	public class HostBuilderExtensionsTests
+	{
+		[DataTestMethod]
+		[DataRow(typeof(IServiceContext), typeof(OmexServiceFabricContext))]
+		[DataRow(typeof(IExecutionContext), typeof(ServiceFabricExecutionContext))]
+		[DataRow(typeof(ITimedScopeProvider), null)]
+		[DataRow(typeof(ILogger<HostBuilderExtensionsTests>), null)]
+		public void AddOmexServiceFabricDependencies_TypesRegistred(Type typeToResolver, Type? expectedImplementationType)
+		{
+			void CheckTypeRegistration<TContext>() where TContext : ServiceContext
+			{
+				object obj = new ServiceCollection()
+					.AddOmexServiceFabricDependencies<StatelessServiceContext>()
+					.AddSingleton(new Mock<IHostEnvironment>().Object)
+					.BuildServiceProvider()
+					.GetService(typeToResolver);
+
+				Assert.IsNotNull(obj, "Failed to resolve for {0}", typeof(TContext));
+
```

Comment:
Should we validate the `expectedImplementationType` is not null

Reply:
It is validated one line bellow

Reply:
I can't see this in the snippet above, but my understanding was that we do a `!= null` check rather than `Assert` it

Reply:
Ah sorry, I've misunderstood you, the idea is that we have different input parameters for this test, it one case we what to make sure that resolved implementation of a certain type. And if we don't care we are just passing `null` for expected type to skip this check.